### ISSUE TITLE
EntityNotAvailableException

### ISSUE THREAD

**Issue by chazt3n** [aws/aws-xray-sdk-dotnet#161]
TL;DR: Am I supposed to make 1000 code changes to actually use xray, or is there an actual way to instrument httpclients, dbcontexts, aws services at a top level of a service without blowing up everywhere?

Do I need to go through my entire app finding places where it will likely explode and introduce change just so I can trace a db context?

HttpClients have very little documentation. In both listed methods they a) Don't have the handler registered with IOC, or b) you can't get the value because some stream has already done something.

The LOG_ERROR doesn't work because the EntityNotAvailableException throws no matter what. In api controllers even with the middleware, in dbcontexts unless I start a segment each time I use one, and I haven't even made it to code that uses AWS SDK yet.

Can I just start a segment on app start and call it good? 

this is my code in `Configure()`
```cs
if (!env.IsDevelopment() || shouldEnableXrayLocally)
{
   AWSXRayRecorder recorder = new AWSXRayRecorderBuilder()
       .WithDaemonAddress(xrayDaemonAddress)
       .WithContextMissingStrategy(ContextMissingStrategy.LOG_ERROR)
       .Build();
    AWSXRayRecorder.InitializeInstance(recorder: recorder);
    AWSSDKHandler.RegisterXRayForAllServices();
    builder.UseXRay(config[ApplicationName]);
}

```

**Comment by chazt3n** [comment]
For reference: new relic experience was pretty OOTB. I fully get that I am causing the problem, but it is super frustrating trying to get past the first steps. 

![image](https://user-images.githubusercontent.com/3843179/99686635-688d4500-2a49-11eb-9a97-1ae6cb5e5723.png)

**Comment by chazt3n** [comment]
I found an issue with how I was setting up our kind of foundational nuget packages for building services.
Within it, it provides an extension method that has a call to `ConfigureServices`. Which works and is _additive._ I just found out that `.Configure()` is not, and last call wins. So xray was being setup and then promptly removed.... Will close this but may need to create another with other issues! Especially the HttpClients, thank you!
